packages/cli/src/plugins/foundry.ts

```diff
@@ -83,23 +86,65 @@ type FoundryConfig<TProject extends string> = {
 
 type FoundryResult = RequiredBy<Plugin, 'contracts' | 'validate' | 'watch'>
 
+const foundryTomlShape = z.object({
+  profile: z.object({
+    default: z.object({
+      out: z.string(),
+    }),
+  }),
+})
+
+const getFoundryConfigSync = (project: string) => {
+  const FOUNDRY_CONFIG_NAME = 'foundry.toml'
+  const path = [project, FOUNDRY_CONFIG_NAME].join('/')
+  let config: unknown
+  try {
+    config = toml.parse(fse.readFileSync(path, 'utf8'))
+  } catch (e) {
+    throw new Error(
+      dedent`
+        Unable to read ${FOUNDRY_CONFIG_NAME} in project
+      `,
+    )
+  }
+  const parsedConfig = foundryTomlShape.safeParse(config)
+  if (!parsedConfig.success) {
+    throw new Error(
+      dedent`
+        Unable to detect artifacts dir for ${FOUNDRY_CONFIG_NAME} at ${pc.gray(
+        project,
+      )}
+        Foundry plugin defaults to profile.default.out property
+        Skip automatic artifact detection by explicitly specifying \`artifacts\` in plugin config
+      `,
+    )
+  }
+  return parsedConfig.data
+}
+
 /**
  * Resolves ABIs from [Foundry](https://github.com/foundry-rs/foundry) project.
  */
-export function foundry<TProject extends string>({
-  artifacts = 'out',
-  deployments = {},
-  exclude = defaultExcludes,
-  forge: {
-    clean = false,
-    build = true,
-    path: forgeExecutable = 'forge',
-    rebuild = true,
-  } = {},
-  include = ['*.json'],
-  namePrefix = '',
-  project,
-}: FoundryConfig<TProject>): FoundryResult {
+export function foundry<TProject extends string>(
+  config: FoundryConfig<TProject>,
+): FoundryResult {
+  const {
+    deployments = {},
+    exclude = defaultExcludes,
+    forge: {
+      clean = false,
+      build = true,
+      path: forgeExecutable = 'forge',
+      rebuild = true,
+    } = {},
+    include = ['*.json'],
+    namePrefix = '',
+    project,
+  } = config
```

Comment:
```suggestion
export function foundry<TProject extends string>({
  artifacts = 'out',
  deployments = {},
  exclude = defaultExcludes,
  forge: {
    clean = false,
    build = true,
    path: forgeExecutable = 'forge',
    rebuild = true,
  } = {},
  include = ['*.json'],
  namePrefix = '',
  project,
}: FoundryConfig<TProject>): FoundryResult {
```

Looking at this now, this didn't need to change.  This is just a relic from my initial implementation